tests/Microsoft.Identity.Web.Test/Resource/AadIssuerValidatorTests.cs

```diff
@@ -0,0 +1,340 @@
+            var jwtSecurityToken = new JwtSecurityToken(issuer: TestConstants.AadIssuer, claims: new[] { issClaim, tidClaim });
+            var expectedErrorMessage = $"Issuer: '{TestConstants.AadIssuer}', does not match any of the valid issuers provided for this application.";
+
+            var exception = Assert.Throws<SecurityTokenInvalidIssuerException>(() => 
+                validator.Validate(TestConstants.AadIssuer, jwtSecurityToken, 
+                    new TokenValidationParameters() { ValidIssuer = TestConstants.B2CIssuer }));
+            Assert.Equal(expectedErrorMessage, exception.Message);
+        }
+
+        [Fact]
+        public void Validate_NotMatchedTenantIds_ThrowsException()
+        {
+            var validator = new AadIssuerValidator(TestConstants.s_aliases);
+            var issClaim = new Claim(TestConstants.ClaimNameIss, TestConstants.AadIssuer);
+            var tidClaim = new Claim(TestConstants.ClaimNameTid, TestConstants.B2CTenantAsGuid);
+            var jwtSecurityToken = new JwtSecurityToken(issuer: TestConstants.AadIssuer, claims: new[] { issClaim, tidClaim });
+            var expectedErrorMessage = $"Issuer: '{TestConstants.AadIssuer}', does not match any of the valid issuers provided for this application.";
+
+            var exception = Assert.Throws<SecurityTokenInvalidIssuerException>(() =>
+                validator.Validate(TestConstants.AadIssuer, jwtSecurityToken,
+                    new TokenValidationParameters() { ValidIssuer = TestConstants.AadIssuer }));
+            Assert.Equal(expectedErrorMessage, exception.Message);
+        }
+
+        [Fact]
+        public void Validate_NotMatchedToMultipleIssuers_ThrowsException()
+        {
+            var validator = new AadIssuerValidator(TestConstants.s_aliases);
+            var issClaim = new Claim(TestConstants.ClaimNameIss, TestConstants.AadIssuer);
+            var tidClaim = new Claim(TestConstants.ClaimNameTid, TestConstants.TenantIdAsGuid);
+            var jwtSecurityToken = new JwtSecurityToken(issuer: TestConstants.AadIssuer, claims: new[] { issClaim, tidClaim });
+            var expectedErrorMessage = $"Issuer: '{TestConstants.AadIssuer}', does not match any of the valid issuers provided for this application.";
+
+            var exception = Assert.Throws<SecurityTokenInvalidIssuerException>(() =>
+                validator.Validate(TestConstants.AadIssuer, jwtSecurityToken,
+                    new TokenValidationParameters()
+                    {
+                        ValidIssuers = new[] {
+                            "https://host1/{tenantid}/v2.0",
+                            "https://host2/{tenantid}/v2.0"
+                        }
+                    }));
+            Assert.Equal(expectedErrorMessage, exception.Message);
+        }
+
+        // Regression test for https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/issues/68
```

Comment:
I left the tests below here from previous version. The main differences is that they pass B2C related info and call GetIssuerValidator instead of the constructor. I think these are duplicates and can be removed. Also I think AadIssuerValidator class itself needs to be refactored; so these tests might change then too.